corpus/filter.py

```diff
@@ -69,6 +71,45 @@ def run(self):
                 )
 
 
+class FilterSegmentsByRegexJob(Job):
+    def __init__(self, segment_files, filter_regex, invert_match=False):
+        """
+        Filters segment list file using a given regular expression
+        :param dict[int,Path] segment_files: original segment list files to be filtered
+        :param str filter_regex: regex used for filtering
+        :param bool invert_match: keep segment if regex does not match (if False) or does match (if True)
+        """
+        self.segment_files = segment_files
+        self.filter_regex = filter_regex
+        self.invert_match = invert_match
+
+        num_segment_lists = len(self.segment_files)
+        self.out_single_segment_files = dict(
+            (i, self.output_path("segments.%d" % i)) for i in range(1, num_segment_lists + 1)
+        )
+        self.out_segment_path = MultiOutputPath(self, "segments.$(TASK)", self.out_single_segment_files)
+
+    def tasks(self):
+        yield Task("run", resume="run", mini_task=True)
+
+    def run(self):
+        pattern = re.compile(self.filter_regex)
+        for idx, segment_file in self.segment_files.items():
+            segment_list = [line.rstrip() for line in open(segment_file.get_path(), "r")]
+            non_empty = False
```

Comment:
```suggestion
            output_empty = True
```
I would try to avoid double negative